src/test/java/org/opensearch/neuralsearch/search/HybridTopScoreDocCollectorTests.java

```diff
@@ -564,4 +564,64 @@ public void testTotalHitsCountValidation_whenTotalHitsCollectedAtTopLevelInColle
         reader.close();
         directory.close();
     }
+
+    @SneakyThrows
+    public void testMaxScoreValidation_whenMaxScoreCollectedAtTopLevelInCollector_thenSuccessful() {
+        final Directory directory = newDirectory();
+        final IndexWriter w = new IndexWriter(directory, newIndexWriterConfig(new MockAnalyzer(random())));
+        FieldType ft = new FieldType(TextField.TYPE_NOT_STORED);
+        ft.setIndexOptions(random().nextBoolean() ? IndexOptions.DOCS : IndexOptions.DOCS_AND_FREQS);
```

Comment:
nit: randomness can introduce flakiness in the test. avoid if possible, seems simple enough to add two separate cases. Try and model your tests to be parametrized so you can easily add cases w/o copy pasting

Reply:
I am following the pattern above which we used earlier to add tests in this Class. Btw, it does not matter if we use random in this case or not. Because the core logic we are testing here is the max score calculated by previous logic and the max score calculated by new logic. If at any point the two scores doesn't match then that will be considered as failure not flakiness.

Reply:
If we have to change it to modularized or maybe optimize the tests then that has to be done in the entire project, Probably we cater it in one complete issue. The score of this PR is to just validate the max score logic.

Reply:
>  am following the pattern above 
I would highly encourage to break anti-patterns when you see one.

> Btw, it does not matter if we use random in this case or not.
Lets remove it then? pick a value

> If we have to change the tests to be parameterized then that has to be done in the entire project
You don't have to revamp all tests. its hard to justify the return on it. You can always start with new tests though.

Reply:
The thing is the place where random() is used it is lucene method. And lucene encourages to use randomized tests https://lucene.apache.org/core/7_3_1/test-framework/org/apache/lucene/util/LuceneTestCase.html . The reason is that it is not possible to cover all the test cases by using parameterized approach. Therefore, lucene recommends Randomized tests to increase more coverage.

Reply:
At any other place which is not a lucene method we have used the paramertized approach for example DOCID1, DOCID2 etc.

Reply:
https://github.com/randomizedtesting/randomizedtesting/wiki/Core-Concepts

Reply:
Therefore, using random() is not a anti pattern.

Reply:
Thanks for the article! 

Randomized testing seems to be having a use case when the complexity of the code is high and its not possible to cover cases everytime. Justifying a random input when it doesn't matter doesn't seems to be the right approach. I will leave it up to you how you want to keep this as it wasn't a blocker in the first place